articles/regularization/index.md

```diff
@@ -0,0 +1,84 @@
+
+Overfitting occurs because a model fails to generalize data that contains a lot of irrelevant data points. Regularization is an answer to overfitting. It is a technique that improves model accuracy as well as prevents loss of important data that is suffered during underfitting. Regularization is a type of regression that shrinks coefficient estimates to zero to reduce the capacity of a model. It removes extra weights from select features and redistributes the weights evenly. This means that regularization discourages the learning of both a model of high complexity and flexibility. In turn, the risk of overfitting is lowered.
+Let’s use a linear regression equation to explain regularization further.
+
+$$ Y = \beta_0 + \beta_1 X_1 + \beta_2 X_2 + … + \beta_p X_p $$
+
+$Y$ represents the value that is to be predicted. $\beta$ stands for the coefficient estimates for various predictors $(X)$. It represents the weights or magnitudes assigned to various predictors.
+A loss function is involved in the fitting process. This loss function is referred to as the residual sum of squares (RSS). Linear models attempt to optimize the coefficients to minimize the loss function. This is shown by the equation for the loss function below.
+
+$$ RSS = \sum\limits_{i=1}^n( y_i – \beta_0 - \sum\limits_{j=1}^p {\beta_j x_ij )^2} $$
+
+Based on the training data, the coefficients shall be adjusted. If there is the existence of noise in the training data, the approximated coefficients shall not generalize well to unseen data. Regularization comes into play and shrinks the learned estimates towards zero.
+
+### Lasso Regression
+
+Lasso regression is a regularization technique to reduce model complexity. It is also known as L1 regularization. Lasso stands for Least Absolute Shrinkage and Selector Operator. Let’s look at the equation below.
+
+$$ \sum\limits_{i=1}^n( y_i – \beta_0 - \sum\limits_{j=1}^p {\beta_j x_ij )^2} + \lambda \sum\limits_{j=1}^p \mid \beta_j \mid = RSS + \lambda \sum\limits_{j=1}^p \mid \beta_j \mid $$
+
+We note that it is a variation of the loss function equation, with the introduction of a penalty term. To penalize high coefficients, lasso uses absolute values of beta $ (\mid \beta\mid) $. Whereas, as we shall learn, ridge regression uses the squares of beta $(\beta^2)$ for a similar purpose.
+Lasso shrinks the regression coefficients to regularize the model parameters. Unlike ridge regression, as we shall cover later, lasso can reduce some regression coefficients to zero. This is particularly important when it comes to feature selection.
+The feature selection process is attributed to the ability of lasso to reduce some regression coefficients to zero. It comes after the regression coefficients are shrunk. Every non-zero regression coefficient is selected for use in the model. Since some features are eliminated, less important features will not feature in the final model. This greatly assists in minimizing prediction errors.
```

Comment:
`The feature selection process is attributed to the ability of lasso to reduce some regression coefficients to zero`. 

You have mentioned this point previously.  Here, you can explain what `Feature selection` is?  and what is the significance when the regression coefficients are reduced to zero?

`It comes after the regression coefficients are shrunk. Every non-zero regression coefficient is selected for use in the model. Since some features are eliminated, less important features will not feature in the final model. This greatly assists in minimizing prediction errors.`

This whole paragraph looks unclear. Please rephrase this paragraph for improving the understandability.